drivers/at86rf2xx/at86rf2xx.c

```diff
@@ -56,6 +56,21 @@ void at86rf2xx_setup(at86rf2xx_t *dev, const at86rf2xx_params_t *params)
     /* initialize device descriptor */
     dev->params = *params;
 #endif
+
+    /* get an 8-byte unique ID to use as hardware address */
+    luid_get(dev->netdev.long_addr, IEEE802154_LONG_ADDRESS_LEN);
+
+    /* make sure we mark the address as non-multicast and not globally unique:
+     * as luid_get() only mutates the first byte, no collisions with other
+     * locally generated LUIDs will arise, if we simply set the two last bits
+     * to its desired value.
+     */
+    dev->netdev.long_addr[IEEE802154_LONG_ADDRESS_LEN - 1] &= ~(0x01);
+    dev->netdev.long_addr[IEEE802154_LONG_ADDRESS_LEN - 1] |=  (0x02);
```

Comment:
This should be the _most_ significant byte, not the least. If you want to change the order internally however, you have to take this into account on `NETOPT` get/set where the order is specified as network byte-order (so msb first).

Reply:
I dropped the ntohl() **and** switched the byte to modify from the first to the last. That way the same byte is modified as before the PR on little endian systems. Either this is now still correct, or you just noticed another bug in the driver. I don't have a 802.15.4 spec at hands to verify. Do you mind to double check?